src/components/MenuBar/ConnectionOptions/ConnectionOptions.test.tsx

```diff
@@ -18,44 +19,87 @@ mockUseAppState.mockImplementation(() => ({ settings: initialSettings, dispatchS
+
     it('should render correctly', () => {
       const wrapper = shallow(<ConnectionOptions />);
       expect(wrapper).toMatchSnapshot();
     });
 
-    it('should dispatch settings changes', () => {
-      const wrapper = shallow(<ConnectionOptions />);
-      wrapper
-        .find(Select)
-        .find({ name: 'dominantSpeakerPriority' })
-        .simulate('change', { target: { value: 'testValue', name: 'dominantSpeakerPriority' } });
-      expect(mockDispatchSetting).toHaveBeenCalledWith({ value: 'testValue', name: 'dominantSpeakerPriority' });
-    });
+    describe('when choose advance setting', () => {
+      it('should dispatch settings changes', () => {
+        const wrapper = shallow(<ConnectionOptions />);
+        wrapper
+          .find(Select)
+          .find({ name: 'dominantSpeakerPriority' })
+          .simulate('change', { target: { value: 'testValue', name: 'dominantSpeakerPriority' } });
+        expect(mockDispatchSetting).toHaveBeenCalledWith({ value: 'testValue', name: 'dominantSpeakerPriority' });
+      });
 
-    it('should not dispatch settings changes from a number field when there are non-digits in the value', () => {
-      const wrapper = shallow(<ConnectionOptions />);
-      wrapper
-        .find(TextField)
-        .find({ name: 'maxTracks' })
-        .simulate('change', { target: { value: '123456a', name: 'maxTracks' } });
-      expect(mockDispatchSetting).not.toHaveBeenCalled();
-    });
+      it('should not dispatch settings changes from a number field when there are non-digits in the value', () => {
+        const wrapper = shallow(<ConnectionOptions />);
+        wrapper
+          .find(TextField)
+          .find({ name: 'maxTracks' })
+          .simulate('change', { target: { value: '123456a', name: 'maxTracks' } });
+        expect(mockDispatchSetting).not.toHaveBeenCalled();
+      });
 
-    it('should dispatch settings changes from a number field when there are only digits in the value', () => {
-      const wrapper = shallow(<ConnectionOptions />);
-      wrapper
-        .find(TextField)
-        .find({ name: 'maxTracks' })
-        .simulate('change', { target: { value: '123456', name: 'maxTracks' } });
-      expect(mockDispatchSetting).toHaveBeenCalledWith({ value: '123456', name: 'maxTracks' });
+      it('should dispatch settings changes from a number field when there are only digits in the value', () => {
+        const wrapper = shallow(<ConnectionOptions />);
+        wrapper
+          .find(TextField)
+          .find({ name: 'maxTracks' })
+          .simulate('change', { target: { value: '123456', name: 'maxTracks' } });
+        expect(mockDispatchSetting).toHaveBeenCalledWith({ value: '123456', name: 'maxTracks' });
+      });
     });
   });
 
   describe('when connected to a room', () => {
     mockUseRoomState.mockImplementation(() => 'connected');
-    it('should render correctly', () => {
+    it('should render correctly and click on ', () => {

```

Comment:
There's no clicking simulation, is that correct?

Reply:
chhanged